src/storage/index/LookUpIndexProcessor.cpp

```diff
@@ -33,22 +33,39 @@ void LookUpIndexProcessor::process(const cpp2::LookUpIndexRequest& req) {
     /**
      * step 3 : execute index scan.
      */
-    for (auto partId : req.get_parts()) {
-        auto code = executeExecutionPlan(partId);
-        if (code != kvstore::ResultCode::SUCCEEDED) {
-            LOG(ERROR) << "Execute Execution Plan! ret = " << static_cast<int32_t>(code)
-                       << ", spaceId = " << spaceId_
-                       << ", partId =  " << partId;
-            if (code == kvstore::ResultCode::ERR_LEADER_CHANGED) {
-                this->handleLeaderChanged(spaceId_, partId);
-            } else {
-                this->pushResultCode(this->to(code), partId);
+    if (executor_ == nullptr) {
+        for (auto partId : req.get_parts()) {
+            auto code = executeExecutionPlan(partId);
+            if (code != kvstore::ResultCode::SUCCEEDED) {
+                LOG(ERROR) << "Execute Execution Plan! ret = " << static_cast<int32_t>(code)
+                           << ", spaceId = " << spaceId_
+                           << ", partId =  " << partId;
+                if (code == kvstore::ResultCode::ERR_LEADER_CHANGED) {
+                    this->handleLeaderChanged(spaceId_, partId);
+                } else {
+                    this->pushResultCode(this->to(code), partId);
+                }
+                this->onFinished();
+                return;
+            }
+        }
+    } else {
+        auto future = executeExecutionPlanConcurrently(req.get_parts());
+        auto map = std::move(future).get();
+        if (!map.empty()) {
```

Comment:
Maybe we don't need to judge whether `map` is empty or not. If map really is empty, `onFinished `is never called.

Reply:
The contents of the `map` are error messages. If the `map` is not empty, it represents an exception during execution, so it returns directly,  and if not,  `onFinished` will be called below.